toolkits/google/arcade_google/tools/contacts.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+from typing import Annotated
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Google
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+
+@tool(requires_auth=Google(scopes=["https://www.googleapis.com/auth/contacts.readonly"]))
+async def search_contacts(
+    context: ToolContext,
+    query: Annotated[
+        str,
+        "The search query for filtering contacts.",
+    ],
+    page_size: Annotated[int, "The maximum number of contacts to return"] = 10,
+) -> Annotated[dict, "A dictionary containing the list of matching contacts"]:
+    """
+    Search the user's contacts using the People API.
+
+    This tool queries the contacts with the provided query string.
+    The API returns contacts that match based on names, email addresses, and more.
+    """
+    # Build the People API service
+    service = build(
+        "people",
+        "v1",
+        credentials=Credentials(
+            context.authorization.token
+            if context.authorization and context.authorization.token
+            else ""
+        ),
+    )
+
+    # Warm-up the cache before performing search.
+    # TODO: Ideally we should warmup only if this user (or google domain?) hasn't warmed up recently
+    _warmup_cache(service)
```

Comment:
Perhaps the warm-up could be implemented as a decorator, which could be aware of the last warm-up time and skip it, in case the index is already warm? not sure if this could be implemented easily considering our cloud engine has to support multi-tenancy... in a single tenant implementation, I imagine even an env var `GOOGLE_CONTACTS_INDEX_LAST_WARMUP_TIMESTAMP` might suffice